app/lib/account/backend.dart

```diff
@@ -176,7 +179,17 @@ class AccountBackend {
     if (auth == null) {
       return null;
     }
-    return await _lookupOrCreateUserByOauthUserId(auth);
+    final user = await _lookupOrCreateUserByOauthUserId(auth);
+    if (user.isDeleted) {
+      // This can only happen if the Google account had been deleted and then
+      // later re-enabled. We don't automatically re-enable such User accounts,
+      // as the new account might be a new owner on the same e-mail address
+      // (e.g. non-Google e-mails on GSuite).
```

Comment:
No, this should only happen if there is a consistency issue, right?

Otherwise, the `oauthUserId` is stripped when `isDeleted` is set, and we never allow upgrading a `User` entity with `isDeleted == true`... right?

Reply:
Ah, yeah, true.